### PR TITLE
feat(da): metrics

### LINKED ISSUE #1126: [EPIC] Metrics
# Overview

We should expose metrics for rollup developers so that they can make informed decisions about their rollup. These metrics could then be viewed in something like a grafance dashboard. 

```[tasklist]
### Tasks
- [ ] https://github.com/rollkit/rollkit/issues/1129
- [ ] Add basic metrics endpoint
- [ ] Add additional metrics
```

Some metrics requested: 

- DA cost/burn rate
- Current DA batch signer’s wallet balance
- What are latest DA bundle submitted, are they successful, etc.

### PR DESCRIPTION
**Pull request by tuxcanfly**
## Overview

Added metrics reporting for DA submission related data. Fixes https://github.com/rollkit/rollkit/issues/1126

## Release Notes

- **New Features**
	- Added support for Prometheus metrics, including new command-line flags for enabling metrics and specifying the metrics server address.
	- Introduced a metrics system for monitoring sequencer performance with various metrics tracked.

- **Documentation**
	- Updated the README.md to include new flags for enabling metrics and specifying the metrics address.

- **Bug Fixes**
	- Enhanced error handling in the sequencer for batch hash validation and recovery.

- **Tests**
	- Updated test cases to accommodate new metrics functionality and increased timeout duration.

### CODE REVIEW

File: sequencing/metrics.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	GasPrice metrics.Gauge
+	// Last submitted blob size
+	LastBlobSize metrics.Gauge
+	// TODO(tuxcanfly): needs gas used, wallet balance from go-da
+	// cost / byte
+	// CostPerByte metrics.Gauge
+	// Wallet Balance
+	// WalletBalance metrics.Gauge
+	// Transaction Status
+	TransactionStatus metrics.Gauge
```
[the remark this answers is not in the document]
Comment (tzdybal) [reply sequencing/metrics.go:42]: Please respond to this @tuxcanfly.

**Review by tzdybal — CHANGES_REQUESTED** [review]
Left some comments.

File: sequencing/metrics.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (tuxcanfly) [reply sequencing/metrics.go:42]: Done 👍🏽

File: sequencing/metrics.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+// PrometheusMetrics returns Metrics build using Prometheus client library.
+// Optionally, labels can be provided along with their values ("foo",
+// "fooValue").
+func PrometheusMetrics(namespace string, labelsAndValues ...string) *Metrics {
+	labels := []string{}
+	for i := 0; i < len(labelsAndValues); i += 2 {
+		labels = append(labels, labelsAndValues[i])
+	}
```
[the remark this answers is not in the document]
Comment (tuxcanfly) [reply sequencing/metrics.go:56]: Done 👍🏽

**Review by gupadhyaya — APPROVED** [review]

**Review by tzdybal — APPROVED** [review]
utACK

**Pull request merged**